third-party/crowdsec/src/main/java/org/apache/james/CrowdsecImapConnectionCheck.java

```diff
@@ -0,0 +1,80 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import static org.apache.james.model.CrowdsecClientConfiguration.DEFAULT_TIMEOUT;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+
+import org.apache.commons.net.util.SubnetUtils;
+import org.apache.james.exception.CrowdsecException;
+import org.apache.james.imap.api.ConnectionCheck;
+import org.apache.james.model.CrowdsecClientConfiguration;
+import org.apache.james.model.CrowdsecDecision;
+import org.apache.james.model.CrowdsecHttpClient;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Mono;
+
+public class CrowdsecImapConnectionCheck implements ConnectionCheck {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CrowdsecImapConnectionCheck.class);
+
+    private final CrowdsecHttpClient client;
+
+    @Inject
+    public CrowdsecImapConnectionCheck(CrowdsecClientConfiguration crowdsecClientConfiguration) {
+        this.client = new CrowdsecHttpClient(crowdsecClientConfiguration);
+    }
+
+    @Override
+    public Publisher<Void> validate(InetSocketAddress remoteAddress) {
+        String ip = remoteAddress.getAddress().getHostAddress();
+
+        return client.getCrowdsecDecisions()
```

Comment:
Cache example: https://github.com/apache/james-project/blob/eff08aed533af2cb79035a05d81e5d698de48979/server/container/guice/configuration/src/main/java/org/apache/james/utils/PropertiesProvider.java#L83

Reply:
IMO caching is overkill.

Connection churn rate is very low for IMAP and SMTP so I expect no noticeable effect on performance. 
Caching brings possible inconsistencies and do not seem like strictly needed.

However I would agree agressively lowering the timeout to say 500ms. If crowdsec do not answer in the given time then it would be ok to me to bypass it. Even better: make the timeout configurable.

Reply:
@thanhbv200585 
if we don't cache it, 
can you check the http api of Crowdsec support is reactive or not?
if yes, the `client.getCrowdsecDecisions()` return `Flux<Result>` is better than `Mono<List<Result>>`

Reply:
Maybe in another ticket?